python-package/lightgbm/plotting.py

```diff
@@ -10,24 +11,36 @@
 from .sklearn import LGBMModel
 
 
-def _check_not_tuple_of_2_elements(obj, obj_name='obj'):
+def _check_not_tuple_of_2_elements(obj: Any, obj_name: str = 'obj') -> None:
     """Check object is not tuple or does not have 2 elements."""
     if not isinstance(obj, tuple) or len(obj) != 2:
         raise TypeError(f"{obj_name} must be a tuple of 2 elements.")
 
 
-def _float2str(value, precision=None):
+def _float2str(value: float, precision: Optional[int] = None) -> str:
     return (f"{value:.{precision}f}"
             if precision is not None and not isinstance(value, str)
             else str(value))
 
 
-def plot_importance(booster, ax=None, height=0.2,
-                    xlim=None, ylim=None, title='Feature importance',
-                    xlabel='Feature importance', ylabel='Features',
-                    importance_type='split', max_num_features=None,
-                    ignore_zero=True, figsize=None, dpi=None, grid=True,
-                    precision=3, **kwargs):
+def plot_importance(
+    booster: Union[Booster, LGBMModel],
+    ax=None,
+    height: Optional[float] = 0.2,
```

Comment:
```suggestion
    height: float = 0.2,
```

This should just be `float`. The `Optional` type in `typing` is not intended to document whether or not something has a default argument. `Optional[some_type]` is just a convenience for saying `Union[some_type, None]`. https://docs.python.org/3/library/typing.html#typing.Union

Any argument which has a default argument for which `None` is not a valid value should not have `Optional` in its hint.